### PR TITLE
Implement EIP-6780: SELFDESTRUCT only in same transaction

### LINKED ISSUE #1461: Update `selfdestruct` according to Cancun upgrade
selfdestruct behaviour changed in the dencun / cancun upgrade. we also have to update our DbCommit implementation to reflect the changes. keep in mind we still have to support activation prior to cancun.

see EIP describing the change https://eips.ethereum.org/EIPS/eip-6780

keep in mind, the same tx code deletion is probably handled by revm itself. but pls double check.

### PR DESCRIPTION
**Pull request by kpp**
# Description

Mostly fixing issues prior to cancun.

## Linked Issues
- Fixes https://github.com/chainwayxyz/citrea/issues/1461

### CODE REVIEW

File: crates/evm/src/evm/handler.rs [hunk 1]
```diff
@@ -485,7 +485,29 @@ fn calc_diff_size<EXT, SPEC: Spec, DB: Database>(
...
-            JournalEntry::AccountDestroyed { address, .. } => {
+            JournalEntry::AccountDestroyed {
+                address,
+                target,
+                was_destroyed,
+                had_balance,
+            } => {
+                // This event is produced only if acc.is_created() || !is_cancun_enabled
+                // State is not changed:
+                // * if we are after Cancun upgrade and
```
Comment (rakanalh) [inline crates/evm/src/evm/handler.rs:496]:
```suggestion
                // * if  Cancun fork is activated and
```

Comment (kpp) [reply crates/evm/src/evm/handler.rs:496]: https://github.com/bluealloy/revm/blob/bbc8d81dbe2a6a4d184e32fa540e1c4e248a65c9/crates/revm/src/journaled_state.rs#L530

Comment (rakanalh) [reply crates/evm/src/evm/handler.rs:496]: Pasting the text as-is doesn't mean that the wording is correct.

**Comment by okkothejawa** [comment]
> Note that if the target is the same as the contract calling SELFDESTRUCT there is no net change in balances. Unlike the prior specification, Ether will not be burnt in this case.

Should this be handled by this PR or does Revm already handle it?

**Comment by kpp** [comment]
> does Revm already handle it?

it does all the magic. Actually it doesnt generate any log in this case so for an observer nothing changes -> no state diff

File: crates/evm/src/evm/handler.rs (hunk 1 above)
Comment (eyusufatik) [reply crates/evm/src/evm/handler.rs:496]: resolving this as both wording work

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**